plasmapy/physics/transport/collisions.py

```diff
@@ -306,11 +309,22 @@ def _boilerPlate(T, particles, V):
     # obtaining reduced mass of 2 particle collision system
     reduced_mass = atomic.reduced_mass(*particles)
 
-    if V == 0:
+    if np.any(V == 0):
         raise utils.exceptions.PhysicsError("You cannot have a collision for zero velocity!")
     # getting thermal velocity of system if no velocity is given
-    if np.isnan(V):
+    if V is None:
         V = parameters.thermal_speed(T, mass=reduced_mass)
+    elif np.any(np.isnan(V)):
+        if np.isscalar(V.value) and np.isscalar(T.value):
+            V = parameters.thermal_speed(T, mass=reduced_mass)
+        elif np.isscalar(V.value):
+            V = parameters.thermal_speed(T, mass=reduced_mass)
+        elif np.isscalar(T.value):
+            V = V.copy()
```

Comment:
:+1:

Reply:
I think there is an implicit assumption here that arrays of temperatures and velocities would be equal in length and produce an array of Coulomb logarithms of equal length. Am I understanding this part correctly?

If so, this should be explicitly documented in the docstring and perhaps add some if statements or try-except blocks to catch when this dimensions don't match.

Reply:
I don't think it should be assumed that T and V are the same length here, this is inside the `np.any(np.isnan(V))`, so V could e.g. be the default scalar nan, while we would still want T to be allowed to be a vector.